### PR TITLE
Issue #36: Add option to install npm/composer/pip/gem packages

### PR DESCRIPTION
**Pull request by oxyc**
Notes:
- As of Ansible 2.4 we can use `global_command` instead of specifying the `working_dir`. 
- Users who want to access Composer bins need to make sure `composer global config bin-dir --absolute` is available in their `$PATH`. https://github.com/geerlingguy/dotfiles/ does this.
- I avoided any complicated module arguments and normalized everything to `item`, `version` and `state`.

Example:

```yaml
composer_packages:
  - hirak/prestissimo
  - name: drush/drush
    version: '^8.1'
```

### CODE REVIEW

**Comment by oxyc** [comment]
I just came across a scenario when I did to use `pip3` rather than`pip`, thus requiring `executable` option for the pip list. Doh...

I still think it might be better to just exclude such options, I can just install it in a separate task. But let me know if you prefer me to add more options to the tasks.

**Comment by geerlingguy** [comment]
Maybe add an undocumented extra variable for the `pip` task, and `default(omit)` it? I'd be game for that.

File: README.md [hunk 1]
```diff
@@ -46,6 +46,15 @@ You can override any of the defaults configured in `default.config.yml` by creat
...
 
+    composer_packages:
+      - hirak/prestissimo
+      - name: drush/drush
+        version: '^8.1'
+
+    gem_packages:
+      - name: bundler
+        state: latest
+
```
Comment (oxyc) [inline README.md]: Or do you prefer if the example is in `default.config.yml`?

Comment (geerlingguy) [inline README.md]: I think having it in both places might be best, like with:

```yaml
gem_packages: []
  # - name: bundler
      state: latest
```

File: default.config.yml [hunk 2]
```diff
@@ -79,9 +79,21 @@ osx_script: "~/.osx --no-restart"
...
+  #   state: present # present/absent/latest, default: present
+  #   version: "~> 1.15.1" # default: N/A
 npm_packages: []
+  # - name: webpack
+  #   state: present # present/absent/latest, default: present
+  #   version: "^2.6" # default: N/A
 pip_packages: []
+  # - name: mkdocs
+  #   state: present # present/absent/latest, default: present
+  #   version: "0.16.3" # default: N/A
```
Comment (oxyc) [inline default.config.yml]: Oh how pleasantly different all these package modules are in Ansible. Even if pip supports version ranges, ansible does not. And then there's the composer modules arguments key...

**Pull request merged**